Made some additions in "Why we need Kubernetes and what can it do" section (#43047)
kubernetes/website: content/en/docs/concepts/overview/_index.md

Kubernetes provides you with:

* **Service discovery and load balancing:**
  Kubernetes can expose a container using the DNS name or using their own IP address.
  If traffic to a container is high, Kubernetes is able to load balance and distribute
  the network traffic so that the deployment is stable.
* **Storage orchestration:**
  Kubernetes allows you to automatically mount a storage system of your choice, such as
  local storages, public cloud providers, and more.
* **Automated rollouts and rollbacks:**
  You can describe the desired state for your deployed containers using Kubernetes,
  and it can change the actual state to the desired state at a controlled rate.
  For example, you can automate Kubernetes to create new containers for your
  deployment, remove existing containers and adopt all their resources to the new container.

> * **Scheduling flexibility and Auto-Scalability:**

Reviewer: ```suggestion
* **Scheduling flexibility and suto-scalability:**
```
For consistency.

  Kubernetes provides you with the flexiblity to schedule the containers
  without writing manual scripts for getting metrices, tracing metrics,

>   different kind of environments (development, production, testing, staging etc.)

Reviewer: ```suggestion
  and supports different environments such as development, production, testing, and staging.
```

>   and helps in scaling the containers up or down, making replica of some pods in other nodes or
>   environment etc. as per your need seemlessly.

Reviewer: ```suggestion
  Moreover, Kubernetes facilitates seamless scaling by automatically replicating pods across nodes or
  environments to meet your specific needs.
```

* **Monitoring:**

>   Kubernetes provides you with the convenience to monitor a large no. of containers,

Reviewer: ```suggestion
  With Kubernetes, monitoring a large number of containers becomes hassle-free.
```

>   and in case of failure of a container/pod it self-heals it or replaces it and notifies you about

Reviewer: ```suggestion
  In the event of a container or pod failure, Kubernetes promptly self-heals or replaces them, notifying you about
```

>   its health check quickly by communicating with API-server and for that you don't need to write
>   "n" no. of manual scripts.

Reviewer: ```suggestion
  their health status through API server communication. This eliminates the requirement of writing
  numerous manual scripts.
```
